test/e2e/tests/eg_upgrade.go

```diff
@@ -0,0 +1,166 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+// +build e2e
+
+package tests
+
+import (
+	"net/url"
+	"os"
+	"testing"
+
+	"fortio.org/fortio/periodic"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/registry"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+var EGUpgradeTest = suite.ConformanceTest{
+	ShortName:   "EGUpgrade",
+	Description: "Upgrading from the last eg version should not lead to failures",
+	Manifests:   []string{"testdata/eg-upgrade.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("Upgrade from an older eg release should succeed", func(t *testing.T) {
+			relName := "eg"
+			depNS := "envoy-gateway-system"
+			lastVersionTag := os.Getenv("last_version_tag")
+			if lastVersionTag == "" {
+				lastVersionTag = "v0.0.0-latest" // Default version tag if not specified
```

Comment:
lets make this `v0.6.0`

Reply:
We can't do a hitless upgrade from v0.6.0, it results in connectivity issue as per:
upgrade completed after 12.14846937s with 41572 requests, 40555 success, 1017 failures and 1017 errors
[7735](https://github.com/envoyproxy/gateway/actions/runs/8235529982/job/22520849948#step:6:7736)

We can do v1.0.0-rc.1 , WDYT?

Reply:
do we know why it failed, this sounds like a release blocker

Reply:
or we can track this as a follow up issue, to unblock this PR

Reply:
also rereading your description, it sounds like upgrade passed, but hitless upgrade failed :) 
I think thats fine for v0.6.0 to `latest`, we can make sure `v1.0.0` to `latest` is hitless
any way you can tweak this PR to make sure some requests succeeded eventually, rather than make sure they are no failures ?

Reply:
yes, maybe we just keep v0.6.0 and i will tweak it to do the requests before the upgrade and after the upgrade. This should always work.

Reply:
awesome thanks !